### PR TITLE
[Enhancement] no need get_dict_code by value just transform directly

### PR DESCRIPTION
**Pull request by zombee0**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/formats/parquet/group_reader.cpp [hunk 1]
```diff
@@ -569,22 +570,54 @@ Status GroupReader::DictFilterContext::rewrite_conjunct_ctxs_to_predicates(
...
+        if (dict_values_after_filter == 0) {
             *is_group_filtered = true;
             return Status::OK();
         }
 
         // ---------
         // get dict codes according to dict values.
-        auto* dict_nullable_column = down_cast<NullableColumn*>(dict_value_column.get());
-        auto* dict_value_binary_column = down_cast<BinaryColumn*>(dict_nullable_column->data_column().get());
+        size_t count_zero = SIMD::count_zero(filter, filter.size() - 1);
```
Comment (DorianZheng) [inline be/src/formats/parquet/group_reader.cpp:587]: seems we don't need to do this, `BatchRunCounter` take care of it.

File: be/src/formats/parquet/group_reader.cpp [hunk 2]
```diff
@@ -569,22 +570,54 @@ Status GroupReader::DictFilterContext::rewrite_conjunct_ctxs_to_predicates(
...
+                        if (filter[index + i]) {
+                            dict_codes[dict_codes_index++] = index + i;
+                        }
+                    }
+                }
+                index += batch.length;
+                batch = count.next_batch();
+            }
+        }
+        bool null_is_ok = filter[filter.size() - 1] == 1;
```
Comment (dirtysalt) [inline be/src/formats/parquet/group_reader.cpp:620]: is null the first or the last value?

Comment (dirtysalt) [reply be/src/formats/parquet/group_reader.cpp:620]: OK. I get it, it's the last value.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by DorianZheng — APPROVED** [review]
LGTM

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by dirtysalt — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**